mcs/class/corlib/System.IO/FileStream.cs

```diff
@@ -188,6 +188,12 @@ internal FileStream (string path, FileMode mode, FileAccess access, FileShare sh
 				throw new ArgumentException ("Name has invalid chars");
 			}
 
+			// IsolatedStorage needs to keep the Name property to the default "[Unknown]"
+			if (!anonymous) {
+				path = Path.InsecureGetFullPath (path);
```

Comment:
I'm not concerned about display. I'm concerned about all the `Directory.Exists (path)`, `Path.GetDirectoryName (path)`, `MonoIO.Open (path,...)` that would still be working with a relative path in the anonymous=true case but get a full path in the other (more common, more tested) case -> source of possible bugs that I think is unnecessary.